### DISCUSSION TITLE
Question regarding unziping vpax

### DISCUSSION THREAD

**Discussion by SharmaAntriksh** [DaxStudio/DaxStudio#990] [General]
I am looking at the vertipaq analyzer by SQLBI and the M code they use contains a lot of Binary functions for unziping .vpax file, which obviously goes over my head, I was wondering how do you do it in DAX Studio? For example I can't access the .ToCardinality property using C#.

![image](https://user-images.githubusercontent.com/73054674/200604658-a77f4b93-0f89-426b-b0a5-60349887392d.png)

**Comment by marcosqlbi** [comment]
The source code for the VertiPaq Analyzer library is available here: https://github.com/sql-bi/VertiPaq-Analyzer/tree/master/src (look at the sample projects we use to test the functions, including VPAX zip/unzip).

In your code you are using the TOM library, which is documented here: https://learn.microsoft.com/en-us/analysis-services/tom/introduction-to-the-tabular-object-model-tom-in-analysis-services-amo?view=asallproducts-allversions
I don't know what you want to do, but the reason why we implemented a library is just because it's hard to do what we want by using M.
